src/typing/type_inference_js.ml

```diff
@@ -5031,11 +5033,13 @@ and function_decl id cx (reason:reason) ~async
   let save_return_exn = Abnormal.swap Abnormal.Return false in
   let save_throw_exn = Abnormal.swap Abnormal.Throw false in
   Flow_js.generate_tests cx reason typeparams (fun map_ ->
+    let type_params_map =
+      type_params_map |> SMap.map (Flow_js.subst cx map_) in
```

Comment:
The type params map is passed down into `mk_body` and may feed into flows, so we need to eliminate the `BoundT`s.